scripts/deploy-prod-sample-module.js

```diff
@@ -0,0 +1,125 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+const fs = require('fs-extra');
+const path = require('path');
+const { argv } = require('yargs');
+
+const {
+  sanitizeEnvVars,
+  nginxOriginStaticsRootDir,
+  npmInstall,
+  npmProductionBuild,
+  getGitSha,
+} = require('./utils');
+
+const {
+  modulePath,
+  skipBuild,
+  skipInstall,
+  bundleStaticsOrigin = 'https://sample-cdn.frank',
+} = argv;
+
+const sanitizedEnvVars = sanitizeEnvVars();
+const nginxOriginStaticsModulesDir = path.resolve(nginxOriginStaticsRootDir, 'modules');
+const originModuleMapPath = path.resolve(nginxOriginStaticsRootDir, 'module-map.json');
+
+const getPreBuiltModuleInfo = (pathToModule) => {
+  const pkgPath = path.resolve(pathToModule, 'package.json');
+  // eslint-disable-next-line import/no-dynamic-require, global-require
+  const { name: moduleName, version: moduleVersion } = require(pkgPath);
+
+  const pathToBundleIntegrityManifest = path.resolve(`${pathToModule}/bundle.integrity.manifest.json`);
+  // eslint-disable-next-line global-require,import/no-dynamic-require
+  const integrityDigests = require(pathToBundleIntegrityManifest);
+
+  return { moduleName, moduleVersion, integrityDigests };
+};
+
+const buildModule = async (pathToModule) => {
+  const pkgPath = path.resolve(pathToModule, 'package.json');
+  // eslint-disable-next-line import/no-dynamic-require, global-require
+  const { name: moduleName, version: moduleVersion } = require(pkgPath);
+
+  if (!skipInstall) {
+    await npmInstall({
+      directory: pathToModule,
+      moduleName,
+      moduleVersion,
+      envVars: sanitizedEnvVars,
+    });
+  }
+
+  await npmProductionBuild({
+    directory: pathToModule,
+    moduleName,
+    moduleVersion,
+    envVars: sanitizedEnvVars,
+  });
+
+  const pathToBundleIntegrityManifest = path.resolve(`${pathToModule}/bundle.integrity.manifest.json`);
+  // eslint-disable-next-line global-require,import/no-dynamic-require
+  const integrityDigests = require(pathToBundleIntegrityManifest);
+
+  return {
+    moduleName, moduleVersion, integrityDigests,
+  };
+};
+
+const deployModuleToProdSampleCDN = async (pathToModule, moduleName) => {
+  const pathToModuleBuildDir = path.resolve(`${pathToModule}/build/`);
+  // use one app git commit sha to namespace modules
+  const gitSha = getGitSha();
+  const pathToOriginModuleStatics = path.resolve(`${nginxOriginStaticsModulesDir}/${gitSha}/${moduleName}`);
+  await fs.ensureDir(pathToOriginModuleStatics);
+  await fs.copy(pathToModuleBuildDir, pathToOriginModuleStatics, { overwrite: true });
+  return pathToOriginModuleStatics;
+};
```

Comment:
Would this update while the local production server is running?

Reply:
yes it will